tests/kernel/threads/dynamic_thread_stack/testcase.yaml

```diff
@@ -0,0 +1,26 @@
+common:
+    tags: kernel security
+    min_ram: 16
+    # ARM MPU is a bit tricky
+    filter: not CONFIG_ARM_MPU
+    # oddly both, qemu_x86 and qemu_riscv work fine
+    platform_exclude: qemu_x86_nommu qemu_x86_virt qemu_x86_tiny
```

Comment:
do not like this level of filtering. Also think about running this without --integration, which will be the case when this lands in main branch, we need to make sure it does not break anything beyond those.

Reply:
Agreed - ideally it would be possible to drill down and figure out what is going on in some of these filtered platforms. I'm not familiar enough with all of them to say what the issues are with certainty

Reply:
Most likely, I'll augment the existing `kernel/threads/thread_stack` test. A separate & simpler test was just more useful when developing the feature.